internal/node/machine_mgr.go

```diff
@@ -344,6 +342,11 @@ func (m *MachineManager) awaitHandshake(vmid string) {
 	for !handshakeOk && !m.stopping() {
 		if time.Now().UTC().After(timeoutAt) {
 			m.log.Error("Did not receive NATS handshake from agent within timeout.", slog.String("vmid", vmid))
+			if len(m.handshakes) == 0 {
+				m.log.Error("First handshake failed, shutting down to avoid inconsistent behavior")
+				_ = m.Stop()
```

Comment:
You can call `m.ctx.Cancel()` here instead of `m.Stop()` and `os.Exit()`.

We need to add a call to `n.shutdown()` as line 124 of internal/node.go.

Reply:
If we merge #137 you can then call `m.cancel()` instead of `m.Stop()`. This will result in the node context cleaning up everything properly 👍🏻